File: backend-java/minimarket-api-java/src/main/java/com/minimarket/api/service/CompanyService.java
package com.minimarket.api.service;

import com.minimarket.api.dto.CompanyDto;
import com.minimarket.api.dto.SaveCompanyDto;
import com.minimarket.api.repository.CompanyRepository;
import com.minimarket.api.repository.ProductRepository;
import com.minimarket.api.util.DtoMapper;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class CompanyService {

    private final CompanyRepository companyRepository;
    private final ProductRepository productRepository;

    public CompanyService(CompanyRepository companyRepository, ProductRepository productRepository) {
        this.companyRepository = companyRepository;
        this.productRepository = productRepository;
    }

    public CompanyDto get() {
        return companyRepository.findById(1).map(DtoMapper::toDto).orElse(null);
    }

    @Transactional
    public ServiceResult<CompanyDto> update(SaveCompanyDto dto) {
        if (dto.businessName() == null || dto.businessName().isBlank()) {
            return ServiceResult.failure("El nombre comercial es obligatorio.");
        }

        if (dto.taxId() == null || dto.taxId().isBlank()) {
            return ServiceResult.failure("El RUC es obligatorio.");
        }

        if (dto.minimumStock() == null || dto.minimumStock() < 0) {
            return ServiceResult.failure("El stock minimo no puede ser negativo.");
        }

        var company = companyRepository.findById(1).orElse(null);
        if (company == null) {
            return ServiceResult.failure("La configuracion de empresa no existe.");
        }

        company.setBusinessName(dto.businessName().trim());
        company.setLegalName(dto.legalName() != null ? dto.legalName().trim() : "");
        company.setTaxId(dto.taxId().trim());
        company.setAddressLine(dto.addressLine() != null ? dto.addressLine().trim() : "");
        company.setPhone(dto.phone() != null ? dto.phone().trim() : "");
        company.setTagline(dto.tagline() != null ? dto.tagline().trim() : "");
        company.setDocumentTitle(dto.documentTitle() != null ? dto.documentTitle().trim() : "");
        company.setCustomerLabel(dto.customerLabel() != null ? dto.customerLabel().trim() : "");
        company.setFooterLine1(dto.footerLine1() != null ? dto.footerLine1().trim() : "");
        company.setFooterLine2(dto.footerLine2() != null ? dto.footerLine2().trim() : "");
        company.setShowTicketPreview(dto.showTicketPreview() != null ? dto.showTicketPreview() : Boolean.TRUE);
        company.setMinimumStock(dto.minimumStock());

        var saved = companyRepository.save(company);

        // El stock minimo es global: sincroniza todos los productos con el nuevo valor.
        productRepository.updateAllMinimumStock(dto.minimumStock());

        return ServiceResult.success(DtoMapper.toDto(saved));
    }
}

File: backend/Minimarket.Api/Services/CompanyService.cs
using Minimarket.Api.DTOs;
using Minimarket.Api.Mapping;
using Minimarket.Api.Repositories;

namespace Minimarket.Api.Services;

public class CompanyService(ICompanyRepository companyRepository, IProductRepository productRepository) : ICompanyService
{
    public async Task<CompanyDto?> GetAsync()
    {
        var company = await companyRepository.GetAsync();
        return company?.ToDto();
    }

    public async Task<(bool Success, string? Error, CompanyDto? Company)> UpdateAsync(SaveCompanyDto dto)
    {
        if (string.IsNullOrWhiteSpace(dto.BusinessName))
            return (false, "El nombre comercial es obligatorio.", null);

        if (string.IsNullOrWhiteSpace(dto.TaxId))
            return (false, "El RUC es obligatorio.", null);

        if (dto.MinimumStock < 0)
            return (false, "El stock minimo no puede ser negativo.", null);

        var company = await companyRepository.GetAsync();
        if (company is null)
            return (false, "La configuracion de empresa no existe.", null);

        company.BusinessName = dto.BusinessName.Trim();
        company.LegalName = dto.LegalName.Trim();
        company.TaxId = dto.TaxId.Trim();
        company.AddressLine = dto.AddressLine.Trim();
        company.Phone = dto.Phone.Trim();
        company.Tagline = dto.Tagline.Trim();
        company.DocumentTitle = dto.DocumentTitle.Trim();
        company.CustomerLabel = dto.CustomerLabel.Trim();
        company.FooterLine1 = dto.FooterLine1.Trim();
        company.FooterLine2 = dto.FooterLine2.Trim();
        company.ShowTicketPreview = dto.ShowTicketPreview;
        company.MinimumStock = dto.MinimumStock;

        companyRepository.Update(company);
        await companyRepository.SaveChangesAsync();

        // El stock minimo es global: sincroniza todos los productos con el nuevo valor.
        await productRepository.UpdateAllMinimumStockAsync(dto.MinimumStock);

        return (true, null, company.ToDto());
    }
}
